evalml/tests/automl_tests/test_automl.py

```diff
@@ -714,8 +714,45 @@ def test_add_to_rankings_no_search(mock_fit, mock_score, dummy_binary_pipeline_c
 
     mock_score.return_value = {'Log Loss Binary': 0.1234}
     test_pipeline = dummy_binary_pipeline_class(parameters={})
-    with pytest.raises(RuntimeError, match="Please run automl"):
-        automl.add_to_rankings(test_pipeline, X, y)
+    assert automl.data_split is None
+
+    automl.add_to_rankings(test_pipeline, X, y)
+    assert isinstance(automl.data_split, StratifiedKFold)
+    assert len(automl.rankings) == 1
+    assert 0.1234 in automl.rankings['score'].values
+
+
+@patch('evalml.pipelines.RegressionPipeline.score')
+@patch('evalml.pipelines.RegressionPipeline.fit')
+def test_add_to_rankings_regression_large(mock_fit, mock_score, dummy_regression_pipeline_class):
+    X = pd.DataFrame({'col_0': [i for i in range(101000)]})
+    y = pd.Series([i for i in range(101000)])
+
+    automl = AutoMLSearch(problem_type='regression', max_time=1, max_iterations=1)
+    test_pipeline = dummy_regression_pipeline_class(parameters={})
+    mock_score.return_value = {automl.objective.name: 0.1234}
+    assert automl.data_split is None
+
+    automl.add_to_rankings(test_pipeline, X, y)
+    assert isinstance(automl.data_split, TrainingValidationSplit)
+    assert len(automl.rankings) == 1
+    assert 0.1234 in automl.rankings['score'].values
+
+
+@patch('evalml.pipelines.RegressionPipeline.score')
+@patch('evalml.pipelines.RegressionPipeline.fit')
```

Comment:
No need to mock `fit` if we're not calling `automl.search`

Do we have separate coverage already for calling `add_to_rankings` after `search` has been called? It would be nice to add that to this test since you're already mocking `fit` :)

Reply:
There are tests that call `search` after `add_to_rankings`. I'll remove the `fit` mocks

Reply:
Would also be nice to make sure `automl.search` behaves properly if `add_to_rankings` is called first as well.